src/main/java/com/vinyl/model/PurchaseProduct.java

```diff
@@ -0,0 +1,116 @@
+package com.vinyl.model;
+
+import java.io.Serializable;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.IdClass;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+import javax.validation.constraints.NotNull;
+
+import com.vinyl.model.PurchaseProduct.PurchaseProductId;
+
+@Entity
+@IdClass(PurchaseProductId.class)
+@Table(name = "PURCHASE_PRODUCT")
+public class PurchaseProduct {
+
+	@NotNull
+	@Column(nullable = false)
+	private Integer nrItems;
+
+	@Id
+	@NotNull
+	@ManyToOne
+	@JoinColumn(name = "purchase_id", nullable = false)
+	private Purchase purchase;
+
+	@Id
+	@NotNull
+	@ManyToOne
+	@JoinColumn(name = "product_id", nullable = false)
+	private Product product;
+
+	public Purchase getPurchase() {
+		return purchase;
+	}
+
+	public void setPurchase(Purchase purchase) {
+		this.purchase = purchase;
+	}
+
+	public Product getProduct() {
+		return product;
+	}
+
+	public void setProduct(Product product) {
+		this.product = product;
+	}
+
+	@SuppressWarnings("serial")
+	public static class PurchaseProductId implements Serializable {
+
+		private Purchase purchase;
+		private Product product;
+
+		public PurchaseProductId(Purchase purchase, Product product) {
+			this.purchase = purchase;
+			this.product = product;
+		}
+
+		public PurchaseProductId() {
+
+		}
+
+		@Override
+		public int hashCode() {
+			final int prime = 31;
+			int result = 1;
+			result = prime * result + ((product == null) ? 0 : product.hashCode());
+			result = prime * result + ((purchase == null) ? 0 : purchase.hashCode());
```

Comment:
was this duplication intended?

Reply:
no.. it was generated automatically this way

Reply:
+ one is product, one is purchase